### DISCUSSION TITLE
Individual postgres: max connections?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#960] [Q&A]
Your announcement today of an individual postgres database is incredible, as that was the one thing that held us back on some projects. Quick question: Do you know what the max connections are to the database? We always have issues with postgres when we run it to power NextJs, as when NextJs creates static pages they connect to the database almost simultaneously  and we get the "Fatal: too many connection" error. Hasura is supposed to do connection pooling out of the box, but I haven't actually seen that it works at all.

**Comment by dbarrosop** [comment]
Our current connection and related memory settings are left to their default values. We will probably tweak them over time as needed but for now these values are a good baseline that should work well for most users. What this means is that `max_connections` should be 115 (15 reserved for superusers and 100 for other users). Having said that, there are a couple of things to take into account here:

1. As you are aware hasura comes with their own connection pooler. By default this connection pooler supports up to 50 different concurrent sessions. There are different values you can tweak here to make the most of this depending on your application. In addition, you should be able to increase the pool to support up to 100 connections.
2. If you are connecting to the database via the connection string you are going through pgbouncer with up to 50 concurrent connections to the database in session mode.

We are quite confident this setup should be good for most applications so it'd be great you could give it a test and let us know how it goes.

If you have doubts/concerns don't hesitate to let us know.

**Comment by osseonews** [comment]
Thanks. Will test. BTW, is Hasura's connection pooling something that needs to be set up or does it work out of the box? I ask only b/c I've never actually seen it work properly. Certainly not 50 concurrent sessions. But, maybe we didn't set it up correctly.

**Reply by dbarrosop** [reply]
It is enabled by default but we tweak the parameters a bit when you create a new app. Feel free to test your app and reach out to us if you feel it isn’t working.
